integration-tests/pkg/executor/executor_k8s.go

```diff
@@ -199,29 +200,35 @@ func (e *K8sExecutor) CapturePodConfiguration(testName, ns, podName string) erro
 	return nil
 }
 
-func (e *K8sExecutor) CaptureNamespaceEvents(testName, ns string) error {
-	events, err := e.clientset.CoreV1().Events(ns).List(context.Background(), metaV1.ListOptions{})
+func (e *K8sExecutor) CreateNamespaceEventWatcher(testName, ns string) (watch.Interface, error) {
+	watcher, err := e.clientset.CoreV1().Events(ns).Watch(context.Background(), metaV1.ListOptions{})
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	logFile, err := common.PrepareLog(testName, ns+"-events.jsonl")
-	if err != nil {
-		return err
-	}
-	defer logFile.Close()
-
-	for _, event := range events.Items {
-		eventJson, err := json.Marshal(event)
+	go func(watcher watch.Interface) {
+		logFile, err := common.PrepareLog(testName, ns+"-events.jsonl")
 		if err != nil {
-			return err
+			fmt.Printf("Failed to open event file: %s\n", err)
+			return
```

Comment:
The log file could be opened outside of the coroutine so that the test could fail if there is a problem with the file creation.